qcodes/dataset/measurements.py

```diff
@@ -190,17 +194,23 @@ def add_result(self, *res_tuple: res_type) -> None:
                 if inserting_this_as_array:
                     raise ValueError("Trying to insert into an ArrayType with "
                                      "a scalar value")
-                if param_spec.type == 'text':
+                if param_spec.type != 'numeric':
                     raise ValueError(f"It is not possible to save a numeric "
                                      f"value for parameter {paramstr!r} "
                                      f"because its type class is "
-                                     f"'text', not 'numeric'.")
+                                     f"{param_spec.type!r}, not 'numeric'.")
             elif isinstance(value, str):
                 if param_spec.type != 'text':
                     raise ValueError(f"It is not possible to save a string "
                                      f"value for parameter {paramstr!r} "
                                      f"because its type class is "
                                      f"{param_spec.type!r}, not 'text'.")
+            elif isinstance(value, (complex, np.complex, np.complex128, np.complex64)):
```

Comment:
shall we define `complex_types = (complex, np.complex, np.complex128, np.complex64)` in, say, `qcodes.math.<some module>` and re-use them everywhere in the code? (i actually wonder why numpy does not have such a definition in the library for the convenience of the humanity...)

Reply:
Moved this to utils and prepared for the other complex pr

Reply:
thanks!